### PR TITLE
refactor: config migration for rly, updating rly CLI commands, UX ali…

### PR DESCRIPTION
**Pull request by tmsdkeys**
- Description:
   - config migration for rly to accommodate provider interface
   -  updating rly CLI commands in the scripts (were outdated)
   - add `-p events` flag to `rly start` command to enable ICA support 
   - UX alignment to give no preferential treatment to either relayer (symmetrical `make init-*` and `make start-*` for both hermes and rly) 
   - update readme for changes
- **Timeout testing did fail for go relayer**, not sure if I'm doing something wrong. Hermes did pick up the packets. Can someone else review?
- closes: #85

### CODE REVIEW

**Comment by tmsdkeys** [comment]
Because the go relayer's `rly tx link [path_name]` sets up client, connection ànd channel and we only need to set up client and connection for ICA, I decided to add an ics20 channel instead. Then the ics27 channel is setup when registering an ICA account. I think it also kind of makes sense to initialize an ics20 channel (even if we're not explicitly using it in this demo)

What do you think? Should we include this (and then also add it for hermes for consistency) or should I instead split up the `rly tx link` command and create client and connection manually?

**Comment by damiannolan** [comment]
Thanks for this @tmsdkeys, would it possible to wait for #101 to add fee enabled channels and then retest this with `rly`?

File: Makefile [hunk 1]
```diff
@@ -132,15 +132,18 @@ init-golang-rly: kill-dev install
...
 
 start: 
 	@echo "Starting up test network"
 	./network/start.sh
 
-start-rly:
+start-hermes:
 	./network/hermes/start.sh
 
+start-golang-rly:
```
Comment (damiannolan) [inline Makefile:144]: We could maybe leave this as `start-rly` for brevity, thoughts? 
I'm happy with either

**Review by damiannolan — APPROVED** [review]
Looks good to me! Thanks @tmsdkeys 🚀

**Comment by tmsdkeys** [comment]
> would it possible to wait for https://github.com/cosmos/interchain-accounts-demo/pull/101 to add fee enabled channels and then retest this with rly?

Sure, was already thinking of adding that at a later time

File: Makefile (hunk 1 above)
Comment (tmsdkeys) [reply Makefile:144]: Yeah, that would be fine by me. So have `start-hermes` and `start-rly` (and same for init). Looks cleaner!

File: network/relayer/interchain-acc-config/rly-start.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# Configure predefined mnemonic pharses
+BINARY=rly
+CHAIN_DIR=./data
+CHAINID_1=test-1
+CHAINID_2=test-2
+RELAYER_DIR=./relayer
+MNEMONIC_1="alley afraid soup fall idea toss can goose become valve initial strong forward bright dish figure check leopard decide warfare hub unusual join cart"
+MNEMONIC_2="record gift you once hip style during joke field prize dust unique length more pencil transfer quit train device arrive energy sort steak upset"
+
+# Ensure rly is installed
```
Comment (seantking) [inline network/relayer/interchain-acc-config/rly-start.sh:12]: nit: ensure golang relayer is installed

**Review by seantking — APPROVED** [review]
Amazing work! Great job.

**Comment by tmsdkeys** [comment]
Thanks for the reviews! 

> Timeout testing did fail for go relayer, not sure if I'm doing something wrong. Hermes did pick up the packets

I found out what was happening, the default block history that is checked for events is 20, but I waited longer than 20 blocks to restart the relayer. Added the `-b` flag to the `rly start` command with value 100 which should be enough for demo purposes.

**Pull request merged**